### PR TITLE
Add unit tests & notFoundForwardName attribute to GetPrincipalPipe

### PR DESCRIPTION
**Pull request by Laurens-makel**
Changes:

- New forward added to `GetPrincipalPipe` which should be taken when the Principal object is null or the principal 's name is either an empty string or null.
- Fully backwards compatible, only if Frank developer configures the `notFoundForwardName;` attribute it will be effective.
- 100% test coverage added.

### CODE REVIEW

**Comment by Laurens-makel** [comment]
Moet herzien worden na het mergen van #6080

**Comment by tnleeuw** [comment]
> Moet herzien worden na het mergen van #6080

Dat PR is nu gemerged dus je kan hieraan werken wanneer je tijd hebt! 😃

**Comment by tnleeuw** [comment]
PS @Laurens-makel : Als het PR klaar is voor review zou je dan Jacobjob, Yannick & mijzelf als reviewers willen toevoegen? Dan krijgen we een seintje om ernaar te kijken.

File: core/src/main/java/org/frankframework/pipes/GetPrincipalPipe.java [hunk 1]
```diff
@@ -33,13 +38,29 @@
  */
 @ElementType(ElementTypes.SESSION)
 public class GetPrincipalPipe extends FixedForwardPipe {
+	private @Getter String notFoundForwardName;
+	protected PipeForward notFoundForward;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+		if (StringUtils.isNotEmpty(getNotFoundForwardName())) {
```
Comment (tnleeuw) [inline core/src/main/java/org/frankframework/pipes/GetPrincipalPipe.java:47]: I often wonder wether it makes more sense to check `isEmpty()` or `isBlank()`.
What do you think is the better check in this case?

**Review by tnleeuw — APPROVED** [review]

File: core/src/main/java/org/frankframework/pipes/GetPrincipalPipe.java (hunk 1 above)
Comment (Laurens-makel) [reply core/src/main/java/org/frankframework/pipes/GetPrincipalPipe.java:47]: As long as I am able to register a forward with a blank name, for example `<Forward name=" " path="doSomething" />`,  I should be able to specify the same value for the `notFoundForwardName` attribute.

**Comment by Laurens-makel** [comment]
@tnleeuw 
Sometimes I review my own work and find some remarks, I've noticed that when `principalName = principal.getName();` returns `null`, this is also returned in the `PipeRunResult`.

I believe, correct me if wrong, this will cause the pipe to output `null` and this behaviour is not really desired? Would it be better to return an empty string?

**Comment by tnleeuw** [comment]
@Laurens-makel Do you want to make more changes to this PR based on your considerations above or shall I merge it as it is now?

File: core/src/test/java/org/frankframework/pipes/GetPrincipalPipeTest.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+		String result=prr.getResult().asString();
+		assertEquals("success", prr.getPipeForward().getName());
+		assertEquals(PRINCIPAL_NAME, result);
+	}
+
+	@Test
+	void getNullPrincipalNameDoPipe() throws Exception {
+		// Given
+		when(principal.getName()).thenReturn(null);
+				when(securityHandler.getPrincipal()).thenReturn(principal);
```
Comment (jkosternl) [inline core/src/test/java/org/frankframework/pipes/GetPrincipalPipeTest.java:103]: Please fix your indentation 😉  except if you confess that you're using Notepad.

**Comment by Laurens-makel** [comment]
> @Laurens-makel Do you want to make more changes to this PR based on your considerations above or shall I merge it as it is now?

I will solve the indentation issues first but no more further changes.

File: core/src/test/java/org/frankframework/pipes/GetPrincipalPipeTest.java (hunk 2 above)
Comment (Laurens-makel) [reply core/src/test/java/org/frankframework/pipes/GetPrincipalPipeTest.java:103]: I'm a fan of notepad++ for anything except writing code.

Not sure when and how this happened, will solve this.

**Pull request merged**